### ISSUE TITLE
Generalizing Date Input

### ISSUE THREAD

**Issue by jungshadow** [uswds/uswds#1920]
## Description
First, thanks for everyone's work on this fantastic project. It's hard to find a set of templates and guidelines that hit all the sweet spots of ease-of-use (for developers), good design, usability, and accessibility.

The date input is one of the few pieces with which I've struggled. I've read through issues #1076, #1108, #860, #751, and #449, but I'm not entirely sure what the roadmap is for the component. Personally, it would be helpful from an architectural standpoint if two things changed:

1. The class name "usa-date-of-birth" made more generalized, such as @carodew's [suggestion to change it to "usa-memorable-date"](https://github.com/18F/web-design-standards/issues/1076#issuecomment-201050417), since it could potentially be reused for other memorable dates.
2. The font-size treatment of the legend was reduced to a smaller size that allows it to fit within a longer identity form.

For number 1, I'm happy to submit a PR with the changes in the **Additional Information** section below (_**NB:** I'm relatively new to the web design standards code, so I'll be more exhaustive in ensuring I have all the disparate pieces before I submit a PR_). For number 2, I'm a little hesitant to propose a change since I'm not a designer and there's a lot of information in a small space (i.e. the legend, the help text, and the individual labels on the text boxes). I'd be happy to submit a change with some guidance.

I hope this is helpful and I'm happy to help, if needed. Thanks, again!

## Additional information [optional]
In `src/html/date-input.html`
```html
<fieldset>
  <legend>Date of birth</legend>
  <span class="usa-form-hint" id="dobHint">For example: 04 28 1986</span>

  <div class="usa-memorable-date">
    <div class="usa-form-group usa-form-group-month">
      <label for="date_of_birth_1">Month</label>
      <input class="usa-input-inline" aria-describedby="dobHint" class="usa-form-control" id="date_of_birth_1" name="date_of_birth_1" pattern="0?[1-9]|1[012]" type="number" min="1" max="12" value="">
    </div>
    ...
  </div>
</fieldset>
```

In `src/stylesheets/components/_forms.scss`
```scss
// Memorable dates

.usa-memorable-date {
  label {
    margin-top: 0;
  }

  [type=number]::-webkit-inner-spin-button,
  [type=number]::-webkit-outer-spin-button {
    -webkit-appearance: none;
    appearance: none;
    margin: 0;
  }

  [type=number] {
    -moz-appearance: textfield;
  }
}
```

**Comment by jungshadow** [comment]
Following some of the conversations on here, it seems as though uswds is going through some fairly substantial revisions. If this issue doesn't apply to the new changes, feel free to close. Thanks!

**Comment by maya** [comment]
Hi @jungshadow, thanks for opening this issue. We've incorporated this in #2595 and should be available in version 2.0.
